gen/gen.go

```diff
@@ -119,23 +147,69 @@ func Gen(exercise string, j interface{}, t *template.Template) error {
 	return ioutil.WriteFile(filepath.Join(dirExercise, "cases_test.go"), src, 0666)
 }
 
-func getPath(jFile string) (jPath, jOrigin, jCommit string) {
+func getLocal(jFile string) (jPath, jOrigin, jCommit string) {
 	// Ideally draw from a .json which is pulled from the official x-common
 	// repository.  For development however, accept a file in current directory
 	// if there is no .json in source control.  Also allow an override in any
 	// case by environment variable.
-	if jPath = os.Getenv("EXTEST"); jPath > "" {
+	if jPath := os.Getenv("EXTEST"); jPath > "" {
 		return jPath, "local file", "" // override
 	}
+
 	c := exec.Command("git", "log", "-1", "--oneline", jFile)
 	c.Dir = dirMetadata
-	ori, err := c.Output()
+	origin, err := c.Output()
 	if err != nil {
 		return "", "local file", "" // no source control
 	}
 	if _, err = os.Stat(filepath.Join(c.Dir, jFile)); err != nil {
 		return "", "local file", "" // not in source control
 	}
+
 	// good.  return source control dir and commit.
-	return c.Dir, "exercism/x-common", string(bytes.TrimSpace(ori))
+	return c.Dir, "exercism/x-common", string(bytes.TrimSpace(origin))
+}
+
+func getRemote(exercise string) (body []byte, jOrigin string, jCommit string, err error) {
+	url := fmt.Sprintf(canonicalDataURL, exercise)
+	resp, err := genClient.Get(url)
+	if err != nil {
+		return []byte{}, "", "", err
+	}
+	if resp.Status != "200 OK" {
```

Comment:
I don't particularly care which of these you do, but there was the possibility of `resp.StatusCode` and `http.StatusOK`